dashboard/src/components/SideMenu/SideMenu.tsx

```diff
@@ -65,6 +46,45 @@ const SideMenu = (): JSX.Element => {
   const notSelectedItemClassName =
     'w-full flex pl-5 py-4 cursor-pointer text-white';
 
+  const useNavigateTo = (path: string): (() => void) => {
+    const router = useRouter();
+    const searchParams = useSearch({ strict: false });
+    const org = searchParams.origin;
+
+    let finalPath = path;
+    if (!finalPath.endsWith('/')) {
+      finalPath = finalPath + '/';
+    }
+
+    return () => {
+      const newPath = `${finalPath}?origin=${org}`;
```

Comment:
please use big variable names, org is probably used more as an abbreviation to organization than to origin